Source/GUI/MainWindow.cpp

```diff
@@ -159,12 +163,15 @@ void MainWindow::makeMemViewer()
           &MainWindow::onOpenMemViewerWithAddress);
 }
 
-void MainWindow::firstHookAttempt()
+void MainWindow::hookIfNotHooked()
 {
-  onHookAttempt();
-  if (DolphinComm::DolphinAccessor::getStatus() ==
+    QTimer::singleShot(3000, this, &MainWindow::hookIfNotHooked);
+
+    if (DolphinComm::DolphinAccessor::getStatus() !=
       DolphinComm::DolphinAccessor::DolphinStatus::hooked)
-    updateMem2Status();
+    {
+      onHookAttempt();
+    }
```

Comment:
Looking at this again, I have some concerns about constantly making a QTimer singleShot. I'll have to profile and test in real world use.

Reply:
After testing this build in real world use, I absolutely want there to be a setting to **Turn Off** auto hooking.
In my own use case I regularly purposely unhook with multiple DMEs up with different freeze values already pre-loaded.

Reply:
Performance wise it's not going to be a concern; it wouldn't be different to a `QTimer`. I wouldn't disagree that a `QTimer` may be a more orthodox approach, though.

Reply:
I think it's a sensible request. I was hoping to add options for disabling auto-hooking once the big **Hook** button is moved to a different place, but I can do it now (or take this out of this PR).

Reply:
It depends if you think its easy enough to implement/add.

I would suggest doing something like this (adding to the settings.ini we generate)
https://github.com/aldelaro5/Dolphin-memory-engine/pull/72/files

Reply:
I'll take auto-hooking out, and open a different PR for it.

Reply:
Commit removed from branch. I will open a different PR shortly (which can then stay on hold until the Qt 6 work is completed).